pkg/tools/tools.go

```diff
@@ -168,7 +168,7 @@ func (t *ToolCall) InvokeTool(ctx context.Context, opt InvokeToolOptions) (any,
 		})
 	}
 
-	return response, nil
+	return response, err
```

Comment:
This is interesting. I am questioning if this was masking any other error in the caller.